nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,386 @@
+import os
+import base64
+import json
+import cgi
+import webbrowser
+import tempfile
+import collections
+
+import numpy as np
+import matplotlib as mpl
+import matplotlib.cm
+
+from .._utils.extmath import fast_abs_percentile
+from .._utils.param_validation import check_threshold
+from .. import datasets, surface
+from . import cm
+
+
+def _get_html_template(template_name='surface_plot_template.html'):
+    template_path = os.path.join(
+        os.path.dirname(__file__), 'data', 'html', template_name)
+    with open(template_path, 'rb') as f:
+        return f.read().decode('utf-8')
+
+
+def add_js_lib(html, embed_js=True):
+    js_dir = os.path.join(os.path.dirname(__file__), 'data', 'js')
+    with open(os.path.join(js_dir, 'surface-plot-utils.js')) as f:
+            js_utils = f.read()
+    if not embed_js:
+        js_lib = """
+        <script
+        src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js">
+        </script>
+        <script src="https://cdn.plot.ly/plotly-gl3d-latest.min.js"></script>
+        <script>
+        {}
+        </script>
+        """.format(js_utils)
+    else:
+        with open(os.path.join(js_dir, 'jquery.min.js')) as f:
+            jquery = f.read()
+        with open(os.path.join(js_dir, 'plotly-gl3d-latest.min.js')) as f:
+            plotly = f.read()
+        js_lib = """
+        <script>{}</script>
+        <script>{}</script>
+        <script>
+        {}
+        </script>
+        """.format(jquery, plotly, js_utils)
+    return html.replace('INSERT_JS_LIBRARIES_HERE', js_lib)
+
+
+class SurfaceView(object):
+    """
+    Represents a web page.
+
+    use str(document) or document.html to get a web page,
```

Comment:
You mean use or uses ?

Can you give overall details why this object is necessary ?

Reply:
use.

it contains the plot, and its method _repr_html_ is used by jupyter notebook for display, and save_as_html and open_in_browser can be used by users